lib/rules/no-deprecated-router-transition-methods.js

```diff
@@ -0,0 +1,190 @@
+'use strict';
+
+const emberUtils = require('../utils/ember');
+const { getImportIdentifier } = require('../utils/import');
+const Stack = require('../utils/stack');
+const types = require('../utils/types');
+const decoratorUtils = require('../utils/decorators');
+
+function getBaseFixSteps(fixer, context, currentClass) {
+  const fixSteps = [];
+  const sourceCode = context.getSourceCode();
+  let serviceInjectImportName = currentClass.serviceInjectImportName;
+  let routerServicePropertyName = currentClass.routerServicePropertyName;
+
+  if (!serviceInjectImportName) {
+    fixSteps.push(
+      fixer.insertTextBefore(
+        sourceCode.ast,
+        "import { inject as service } from '@ember/service';\n"
+      )
+    );
+
+    serviceInjectImportName = 'service';
+  }
+
+  if (!routerServicePropertyName) {
+    fixSteps.push(
+      fixer.insertTextBefore(
+        currentClass.node.body.body[0],
+        `@${serviceInjectImportName}('router') router;\n`
+      )
+    );
+
+    routerServicePropertyName = 'router';
+  }
+
+  return { fixSteps, routerServicePropertyName };
+}
+
+const ERROR_MESSAGE =
+  'Do not rely on implicit service injections, these were deprecated in Ember 3.26 and will not work in 4.0.';
```

Comment:
Can we make this error message more specific to the deprecation of implicit router transition methods?

Reply:
I've updated the error message to create a more descriptive and matching message based on usage. The error before was a bad copy/paste from my other rule.